src/components/Drawer/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import { BackDrop, Content, Wrapper } from './styles';
+import { DrawerProps } from './types';
+import Portal from '../Portal';
+
+export default function Drawer({
+  type = 'left',
+  open,
+  onClose,
+  children,
+}: DrawerProps) {
+  const [toggle, setToggle] = useState(false);
+
+  useEffect(() => {
+    if (open) {
+      document.body.style.overflow = 'hidden';
+      setTimeout(() => {
+        setToggle((prev) => !prev);
+      }, 250);
```

Comment:
Remove magic number, and don't use `setTomeout`.\
Check `animationend` event instead.